work/load/load_processors.go

```diff
@@ -0,0 +1,175 @@
+package load
+
+import (
+	"context"
+	"fmt"
+	"math/rand/v2"
+	"slices"
+	"time"
+
+	"github.com/tidepool-org/platform/errors"
+	"github.com/tidepool-org/platform/pointer"
+	work "github.com/tidepool-org/platform/work"
+)
+
+const (
+	quantity                     = 1
+	frequency                    = 5 * time.Second
+	processingAvailableTimeDelay = 5 * time.Second
+	processingTimeout            = 5 * 60
+
+	MetadataProcessResult = "processResult"
+)
+
+func domainName(subdomain string) string {
+	return fmt.Sprintf("org.tidepool.work.test.load.%s", subdomain)
+}
+
+// TypeSleepy - will sleep for a random amount of time from 0 - `sleepMaxMillis` and then returns `ResultDelete`
+var TypeSleepy = domainName("sleepy")
+
+// TypeDopey - has to be told what to do or by default returns `ResultDelete`
+var TypeDopey = domainName("dopey")
+
+type loadProcessor struct {
+	pType string
+}
+
+func newLoadProcessor(pType string) (*loadProcessor, error) {
+	if pType != TypeSleepy && pType != TypeDopey {
+		return nil, errors.Newf("type %s is invalid. Must be one of [%s,%s]", pType, TypeSleepy, TypeDopey)
+	}
+	return &loadProcessor{
+		pType: pType,
+	}, nil
+}
+
+func (p *loadProcessor) Type() string {
+	return p.pType
+}
+
+func (p *loadProcessor) Quantity() int {
+	return quantity
+}
+
+func (p *loadProcessor) Frequency() time.Duration {
+	return frequency
+}
+
+func (p *loadProcessor) getProcessResult(wrk *work.Work, result any, errMsg *string) *work.ProcessResult {
+	switch result.(string) {
+	case work.ResultFailing:
+
+		msg := "failing from load test"
+		if errMsg != nil {
+			msg = *errMsg
+		}
+
+		return work.NewProcessResultFailing(work.FailingUpdate{
+			FailingError: *errors.NewSerializable(errors.New(msg)),
+			Metadata:     wrk.Metadata,
+		})
+	case work.ResultFailed:
+		return work.NewProcessResultFailed(work.FailedUpdate{
+			FailedError: *errors.NewSerializable(errors.New("failed from load test")),
+			Metadata:    wrk.Metadata,
+		})
+	case work.ResultSuccess:
+		return work.NewProcessResultSuccess(work.SuccessUpdate{
+			Metadata: wrk.Metadata,
+		})
+	default:
+		return work.NewProcessResultDelete()
+	}
+}
+
+func (p *loadProcessor) chooseDopeyResult(wrk *work.Work) string {
+	possibleWorkResults := []string{
+		work.ResultDelete,
+		work.ResultFailing,
+		work.ResultFailed,
+		work.ResultSuccess,
+	}
+	if result := wrk.Metadata[MetadataProcessResult]; result != nil {
+		if resultStr, ok := result.(string); ok {
+			if resultStr == "random" {
+				var indexnum int = rand.IntN(len(possibleWorkResults) - 1)
+				return possibleWorkResults[indexnum]
+			}
+			if slices.Contains(possibleWorkResults, resultStr) {
+				return resultStr
+			}
+		}
+	}
+	return work.ResultDelete
```

Comment:
Rather than defaulting to delete, you could default to random. Then no need for the "random" check above. If `MetadataProcessResult` isn't specified then that means random.